### PR TITLE
Orm events refactoring

### PR DESCRIPTION
**Pull request by dshulyak**

### CODE REVIEW

File: solar/solar/interfaces/orm.py [hunk 1]
```diff
@@ -477,6 +483,40 @@ def add_input(self, name, schema, value):
...
+    parent = db_field(schema='str!')
+    parent_action = db_field(schema='str!')
+    etype = db_field('str!')
+    state = db_field('str')
+    child = db_field('str')
+    child_action = db_field('str')
+
+    @classmethod
+    def load_list(cls, parent):
+        rs = db.all(collection=cls._collection.name + ':' + parent)
```
Comment (CGenie) [inline solar/solar/interfaces/orm.py]: Why are you making keys that don't exist. The `_collection` parameter is part of enum and should stay this way. If you want relation to resource, use the `DBRelatedField`.

File: solar/solar/interfaces/db/redis_graph_db.py [hunk 2]
```diff
@@ -152,6 +152,11 @@ def get(self, name, collection=BaseGraphDB.DEFAULT_COLLECTION,
         except TypeError:
             raise KeyError
 
+    def delete(self, name, collection=BaseGraphDB.DEFAULT_COLLECTION):
```
Comment (CGenie) [inline solar/solar/interfaces/db/redis_graph_db.py]: This is not complete -- you need to add `abstractmethod` to `db/base.py`

File: solar/solar/interfaces/orm.py [hunk 3]
```diff
@@ -477,6 +483,40 @@ def add_input(self, name, schema, value):
...
+        return [cls(**r.properties) for r in rs]
+
+    @classmethod
+    def delete_list(cls, parent):
+        db.delete('*', collection=cls._collection.name + ':' + parent)
```
Comment (CGenie) [inline solar/solar/interfaces/orm.py]: I would make a `delete_all` method in `db` not to add strange things as `*` here. Also, same remark about `_collection.name + ':' + parent` as above.

File: solar/solar/interfaces/orm.py [hunk 4]
```diff
@@ -477,6 +483,40 @@ def add_input(self, name, schema, value):
...
+
+    @property
+    def _db_key(self):
+        if not self._primary_field.value:
+            setattr(self, self._primary_field.name,
+                    '{}:{}:{}:{}:{}'.format(
+                        self.parent, self.parent_action,
+                        self.state, self.child, self.child_action))
+            self._update_fields_values()
+        return DBObject._db_key.fget(self)
```
Comment (CGenie) [inline solar/solar/interfaces/orm.py]: What is `fget`?

File: solar/solar/events/controls.py [hunk 5]
```diff
@@ -91,15 +94,15 @@ class React(Event):
 
     def insert(self, changed_resources, changes_graph):
 
-        if self.parent in changes_graph:
-            if self.dependent not in changes_graph:
+        if self.parent_node in changes_graph:
+            if self.child_node not in changes_graph:
                 changes_graph.add_node(
-                    self.dependent, status='PENDING',
+                    self.child_node, status='PENDING',
```
Comment (CGenie) [inline solar/solar/events/controls.py]: You could use the enum here. Raw strings in such cases are sometimes sources of subtle bugs.

File: solar/solar/events/controls.py [hunk 6]
```diff
@@ -91,15 +94,15 @@ class React(Event):
...
                     errmsg=None, type='solar_resource',
```
Comment (CGenie) [inline solar/solar/events/controls.py]: Enum for `solar_resource`

File: solar/solar/events/controls.py (hunk 5 above)
Comment (dshulyak) [inline solar/solar/events/controls.py]: i changed it already in separate request, will rebase that request after this one will be merged

**Comment by loles** [comment]
+1

**Pull request merged**